### PR TITLE
Fix/ballot 1.0.0 issue 360

### PR DESCRIPTION
**Pull request by ryma2fhir**
changed UK extension to be an addition to the existing International extension just for the description text part. https://hl7.org/FHIR/extension-coding-sctdescid.html 

Kept the name the same as the international extension. I am wondering if the extension text should reference that it is an extension of the international one?

Is this still an extension of the Coding datatype, or should it reference the International extension-coding-sctdescid instead within the IG?

@SheenaSharma

### CODE REVIEW

**Review by declankieran-nhsd — CHANGES_REQUESTED** [review]
Yes I think you're right, it should be referencing the coding-sctdescid extension.

Options here seem to be having just an extension that provides the display.  The name of the extension is Extension-UKCore-CodingSCT**DescId**.  So something to think about here would be whether this needs to be changed to something like Extension-UKCore-CodingSCT**Display** or possibly Extension-UKCore-CodingSCTDescId-**Display** to indicate this should be used in combination with coding-sctdescid.

An example of doing this is given here https://github.com/NHSDigital/fhir-transforms/pull/42#issuecomment-1062912121

However, having these separate doesn't seem to be that sensible though.  Arguably you would want to always have the SCTDescId value being included when using the display value, so maybe this extension should include a Modifier Extension that requires the use of the coding-sctdescid within the Extension-UKCore-CodingSCTDescId-**Display** (*name changed*) extension.

The example in the other repo shows where the extension just provides the display, but is done within the context of a CodeableConcept, which I think would be the case with the coding-sctdescid extension as well, so being separate might not be so much of an issue, but probably worth talking about.

Also, this change has broken (at least) this example

https://github.com/NHSDigital/FHIR-R4-UKCORE-STAGING-MAIN/blob/fix/Ballot-1.0.0_Issue-360/examples/UKCore-Condition-Extension-CodingSCTDescId-Example.xml

So this will need changed to reflect whatever change is made.

**Comment by ryma2fhir** [comment]
To be documented and taken to the DA to decide route to take regarding the referencing

**Comment by ryma2fhir** [comment]
The DA agreed that this extension should estend the international extension coding-sctdescid.

This has been done and passed, would like someone else to run through and ensure it looks correct. 

---
Another problem with this is it possibly contradicts ballot issue #62 

Existing Wording
> "CodeableConcept.text: 
> Only populate when there is no user selected translation set with a display or descriptionDisplay recorded."

Ballot comment (from Epic)
> Is it a recommendation or a requirement to omit CodeableConcept.text when a user selection display / descriptionDisplay is present? 
> 
> It is not uncommon for text to be populated even if it is a duplicate of a coding's display, simply for consistency in formatting. This is also endorsed in the FHIR specifications: https://www.hl7.org/fhir/datatypes-definitions.html#CodeableConcept.text
> 
> "Very often the text is the same as a displayName of one of the codings."

DigiMeds Feedback

> remove the constraint on this element

If we do split it then it works regarding FHIR principles, and I don't understand why you need to duplicate things, but it may cause problems elsewhere

File: structuredefinitions/Extension-UKCore-CodingSCTDesc.xml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+	 <slicing>
+	 <discriminator>
+	<type value="value"/>
+	<path value="url"/>
+	</discriminator>
+	<rules value="open"/>
+	</slicing>
+	<min value="1"/>
+	</element>
+	 <element id="Extension.extension:descriptionDisplay">
```
Comment (declankieran-nhsd) [inline structuredefinitions/Extension-UKCore-CodingSCTDesc.xml:65]: I dont think you need to slice here, I think using the same structure as the international extension **coding-sctdescid** is probably what we want here, as the display will be used in the same way.  See here for the StructureDefinition of **coding-sctdescid** https://hl7.org/fhir/r4/extension-coding-sctdescid.json.html

File: structuredefinitions/Extension-UKCore-CodingSCTDesc.xml [hunk 2]
```diff
@@ -0,0 +1,93 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (declankieran-nhsd) [inline structuredefinitions/Extension-UKCore-CodingSCTDesc.xml:1]: XML isn't indented properly, maybe run it through a pretty print after.

**Comment by declankieran-nhsd** [comment]
An example (possibly redoing the examples using the old extension>) once you've finished the new extension would be good, both for content and running through validation.

**Comment by declankieran-nhsd** [comment]
> Another problem with this is it possibly contradicts ballot issue #62
> 
> Existing Wording:
> 
> > "CodeableConcept.text:
> > Only populate when there is no user selected translation set with a display or descriptionDisplay recorded."
> 
> Ballot comment (from Epic):
> 
> > Is it a recommendation or a requirement to omit CodeableConcept.text when a user selection display / descriptionDisplay is present?
> > It is not uncommon for text to be populated even if it is a duplicate of a coding's display, simply for consistency in formatting. This is also endorsed in the FHIR specifications: https://www.hl7.org/fhir/datatypes-definitions.html#CodeableConcept.text
> > "Very often the text is the same as a displayName of one of the codings."
> 
> DigiMeds Feedback:
> 
> > remove the constraint on this element
> 
> If we do split it then it works regarding FHIR principles, and I don't understand why you need to duplicate things, but it may cause problems elsewhere

Is this not solved if we use the new extension? i.e. it will just be base constraints on CodeableConcept.text

File: structuredefinitions/Extension-UKCore-CodingSCTDesc.xml (hunk 1 above)
Comment (ryma2fhir) [reply structuredefinitions/Extension-UKCore-CodingSCTDesc.xml]: > ink using t

I htought that, but was using the old UKCore SCTDescid as a base. Now reworked to use the FHIR version structure

**Review by declankieran-nhsd — APPROVED** [review]

**Pull request merged**